### DISCUSSION TITLE
Add integration test environment to Medusa backend

### DISCUSSION THREAD

**Discussion by vholik** [medusajs/medusa#6316] [Feature Requests]
Currently, Medusa only supports a unit test environment, limiting our ability to thoroughly test security and database integration, especially in scenarios where Medusa is extended as a marketplace. To address this, I suggest adding an integration test environment to enhance testing capabilities.

**Comment by chrismacp** [comment]
Some more information on setting up testing environments within Medusa would be an awesome addition to the docs. 

I kept coming across Docker references and test utilities like in this [test database helper][1] only to see they were all removed. What really impresses me and improves my ability to get up to speed and work quickly with a framework is the good docs around setting up your environment which I feel is lacking here to some degree. A default set up for a new project would be even better, like the starter project seems to have had before. 





[1]: https://github.com/riqwan/medusa-store/blob/main/src/test-helpers/test-database.ts
